kore/src/Kore/Strategies/ProofState.hs

```diff
@@ -6,21 +6,39 @@ module Kore.Strategies.ProofState
     ( extractGoalRem
     , extractUnproven
     , ProofState (..)
+    , Prim (..)
     , proofState
     , ProofStateTransformer (..)
     ) where
 
 import Data.Hashable
 import GHC.Generics as GHC
 
+
+{- | The primitive transitions of the all-path reachability proof strategy.
+ -}
+data Prim rule
+    = CheckProven
+    -- ^ End execution on this branch if the state is 'Proven'.
+    | CheckGoalRemainder
+    -- ^ End execution on this branch if the state is 'GoalRemainder'.
+    | ResetGoal
+    -- ^ Mark all goals rewritten previously as new goals.
+    | Simplify
+    | RemoveDestination
+    | TriviallyValid
+    | DerivePar [rule]
+    | DeriveSeq [rule]
+    deriving (Show)
+
 {- | The state of the all-path reachability proof strategy for @goal@.
  -}
-data ProofState goal
-    = Goal goal
+data ProofState a
```

Comment:
why did `goal` became `a`?

Reply:
I did this to keep the notation similar to the one used in `Goal` (in the `OnePathRule` instance the `goal` type only determines that we will use `ProofState.ProofState a`, for any `a`). If you think this is unnecessary/confusing I can change it back.